driver-sync/src/test/functional/com/mongodb/client/model/mql/AggregatesSearchTest.java

```diff
@@ -0,0 +1,269 @@
+                                + "   { _id: 6, description: 'desc 1' },\n"
+                                + "]");
+    }
+
+    @Test
+    public void testWildcard() {
+        List<Bson> pipeline = Arrays.asList(
+                search(SearchOperator.wildcard(fieldPath("description"), "desc*"),
+                        searchOptions().index(searchIndexName)));
+        assertResults(pipeline, "[\n"
+                                + "   { _id: 6, description: 'desc 1' },\n"
+                                + "   { _id: 7, description: 'desc 8' },\n"
+                                + "]");
+    }
+
+    @Test
+    public void testPhrase() {
+        List<Bson> pipeline = Arrays.asList(
+                search(SearchOperator.phrase(fieldPath("summary"), "one five").slop(2),
+                        searchOptions().index(searchIndexName)));
+        assertResults(pipeline, "[\n"
+                                + "   { _id: 8, summary: 'summary 1 one five' },\n"
+                                + "]");
+    }
+
+    @Test
+    public void testQueryString() {
+        List<Bson> pipeline = Arrays.asList(
+                search(SearchOperator.queryString(fieldPath("summary"), "summary: one AND summary: three"),
+                        searchOptions().index(searchIndexName)));
+        assertResults(pipeline, "[\n"
+                                + "   { _id: 9, summary: 'summary 2 one two three four five' },\n"
+                                + "]");
+    }
+
+    private static void insertDocuments(final String s) {
+        List<Document> documents = BsonArray.parse(s).stream()
+                .map(v -> new Document(v.asDocument()))
+                .collect(Collectors.toList());
+        collection.insertMany(documents);
+    }
+
+    private static void assertResults(final List<Bson> pipeline, final String expectedResultsAsString) {
+        ArrayList<Bson> pipeline2 = new ArrayList<>(pipeline);
+        pipeline2.add(sort(ascending("_id")));
+
+        List<BsonDocument> expectedResults = parseToList(expectedResultsAsString);
+        List<BsonDocument> actualResults = aggregate(pipeline2);
+        assertEquals(expectedResults, actualResults);
+    }
+
+    private static List<BsonDocument> aggregate(final List<Bson> stages) {
+        AggregateIterable<Document> result = collection.aggregate(stages);
+        List<BsonDocument> results = new ArrayList<>();
+        result.forEach(r -> results.add(r.toBsonDocument()));
+        return results;
+    }
+
+    public static List<BsonDocument> parseToList(final String s) {
+        return BsonArray.parse(s).stream().map(v -> toBsonDocument(v.asDocument())).collect(Collectors.toList());
+    }
+
+    public static BsonDocument toBsonDocument(final BsonDocument bsonDocument) {
+        return getDefaultCodecRegistry().get(BsonDocument.class).decode(bsonDocument.asBsonReader(), DecoderContext.builder().build());
+    }
+
+    @Nullable
+    private static Document indexRecord(
```

Comment:
i initially thought this method also indexes some record. I suggest we rename to `find...()`: 
```suggestion
    private static Document findSearchIndex(
```